backend/src/od.rs

```diff
@@ -0,0 +1,148 @@
+use geo::{BoundingRect, Contains, MultiPolygon, Point};
+use geojson::GeoJson;
+use nanorand::{Rng, WyRand};
+use serde::{Deserialize, Serialize};
+use utils::Mercator;
+
+use crate::{IntersectionID, MapModel};
+
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq, PartialOrd, Ord, Serialize, Deserialize)]
+pub struct ZoneID(pub usize);
+
+/// Origin/destination demand data, representing driving trips between two zones.
+#[derive(Serialize, Deserialize)]
+pub struct DemandModel {
+    pub zones: Vec<Zone>,
+    // (zone1, zone2, count), with count being the number of trips from zone1 to zone2
+    pub desire_lines: Vec<(ZoneID, ZoneID, usize)>,
+}
+
+impl DemandModel {
+    /// Turn all of the zones into Mercator. Don't do this when originally building and serializing
```

Comment:
The zone data is shipped in as WGS84, then turned into Mercator for use here. I thought about just doing this upfront and serializing as Mercator, but it's too risky, since we calculate the bounding box from OSM input and turn into mercator here. That probably differs from the bbox around the study area GJ boundaries.